# ğŸ‘‘ Reglas de AdministraciÃ³n

## ğŸ“ Reglas Generales

### GestiÃ³n de Usuarios
1. Niveles de AdministraciÃ³n
   - Super Admin
   - Admin
   - Moderador
   - Soporte
   - Auditor

2. Permisos por Nivel
   - ConfiguraciÃ³n del sistema
   - GestiÃ³n de usuarios
   - ModeraciÃ³n de contenido
   - Reportes y anÃ¡lisis
   - Soporte al cliente

### ModeraciÃ³n
1. Contenido
   - RevisiÃ³n de productos
   - ValidaciÃ³n de imÃ¡genes
   - Control de descripciones
   - Filtro de lenguaje

2. Subastas
   - VerificaciÃ³n de legitimidad
   - Control de precios
   - ValidaciÃ³n de categorÃ­as
   - Monitoreo de actividad

## ğŸ›¡ï¸ Acciones Administrativas

### Usuarios
```typescript
interface UserActions {
    suspend(userId: string, reason: string): void;
    ban(userId: string, reason: string): void;
    restrict(userId: string, restrictions: string[]): void;
    verify(userId: string, level: string): void;
}
```

### Contenido
```typescript
interface ContentActions {
    approve(contentId: string): void;
    reject(contentId: string, reason: string): void;
    flag(contentId: string, type: string): void;
    feature(contentId: string, duration: number): void;
}
```

## ğŸ¯ ModeraciÃ³n de Contenido

### RevisiÃ³n Manual
1. PriorizaciÃ³n
   - Reportes de usuarios
   - Contenido flaggeado
   - Nuevos usuarios
   - Montos altos

2. Proceso
   - RevisiÃ³n inicial
   - VerificaciÃ³n detallada
   - DecisiÃ³n
   - NotificaciÃ³n
   - Seguimiento

### AutomatizaciÃ³n
1. Filtros
   - Palabras prohibidas
   - ImÃ¡genes inapropiadas
   - Enlaces maliciosos
   - Spam detection

2. Machine Learning
   - DetecciÃ³n de fraude
   - AnÃ¡lisis de patrones
   - PredicciÃ³n de riesgos
   - CategorizaciÃ³n

## ğŸ“Š Reportes y AnÃ¡lisis

### MÃ©tricas Clave
1. Usuarios
   - Crecimiento
   - RetenciÃ³n
   - ConversiÃ³n
   - SatisfacciÃ³n

2. Contenido
   - Calidad
   - Engagement
   - Reportes
   - ModeraciÃ³n

3. Transacciones
   - Volumen
   - Valor
   - Ã‰xito
   - Disputas

### Dashboard
```typescript
interface AdminDashboard {
    realTimeMetrics: {
        activeUsers: number;
        ongoingAuctions: number;
        pendingModeration: number;
        recentReports: number;
    };
    
    dailyStats: {
        newUsers: number;
        completedAuctions: number;
        moderationActions: number;
        revenue: number;
    };
}
```

## ğŸ”„ Flujos de Trabajo

### ModeraciÃ³n
```
[Reporte Recibido] â†’ [AsignaciÃ³n] â†’ [RevisiÃ³n] â†’ [DecisiÃ³n]
         â†“                                            â†“
    [Auto-resuelto]                             [ApelaciÃ³n]
```

### Soporte
```
[Ticket Creado] â†’ [ClasificaciÃ³n] â†’ [AsignaciÃ³n] â†’ [ResoluciÃ³n]
        â†“                                             â†“
   [Auto-respuesta]                              [Seguimiento]
```

## ğŸš« Restricciones

### Acciones CrÃ­ticas
1. Requieren aprobaciÃ³n mÃºltiple
   - Baneos permanentes
   - Cambios de sistema
   - ConfiguraciÃ³n global
   - Acciones masivas

2. Logging obligatorio
   - QuiÃ©n
   - QuÃ©
   - CuÃ¡ndo
   - Por quÃ©

### Tiempos de Respuesta
1. CrÃ­tico: < 1 hora
   - Fraude detectado
   - Problema de seguridad
   - CaÃ­da de sistema
   - Disputa grave

2. Alto: < 4 horas
   - Reportes urgentes
   - Problemas de pago
   - Disputas activas
   - Bugs crÃ­ticos

3. Normal: < 24 horas
   - ModeraciÃ³n regular
   - Soporte general
   - Verificaciones
   - Consultas

## ğŸ“± Notificaciones Admin

### Urgentes
1. Seguridad
   - Intentos de hack
   - Accesos sospechosos
   - Fallos crÃ­ticos
   - Alertas de fraude

2. Sistema
   - CaÃ­das
   - Errores graves
   - LÃ­mites alcanzados
   - Problemas de rendimiento

### Rutinarias
1. Diarias
   - Resumen de actividad
   - Tareas pendientes
   - KPIs principales
   - Alertas activas

2. Semanales
   - AnÃ¡lisis de tendencias
   - Reportes de rendimiento
   - Problemas recurrentes
   - Recomendaciones

## ğŸ“Š AuditorÃ­a

### Registro
1. Acciones administrativas
   - Timestamp
   - Usuario admin
   - AcciÃ³n realizada
   - Detalles completos

2. Cambios de sistema
   - ConfiguraciÃ³n anterior
   - Nueva configuraciÃ³n
   - RazÃ³n del cambio
   - Aprobadores

### Reportes
1. Actividad administrativa
   - Por admin
   - Por tipo de acciÃ³n
   - Por resultado
   - Por perÃ­odo

2. Efectividad
   - Tiempo de resoluciÃ³n
   - SatisfacciÃ³n
   - PrecisiÃ³n
   - Consistencia
