# ğŸ”¨ Reglas de Subastas

## ğŸ“ Reglas Generales

### CreaciÃ³n de Subastas
1. Solo usuarios verificados pueden crear subastas
2. El producto debe estar completamente descrito
3. Debe tener al menos una imagen
4. El precio inicial debe ser mayor que 0
5. La fecha de inicio debe ser futura
6. La duraciÃ³n debe estar entre 1 y 30 dÃ­as

### Precio y Pujas
1. El incremento mÃ­nimo debe ser al menos 1% del precio actual
2. El depÃ³sito requerido debe estar entre 5% y 20%
3. No se permiten pujas en los Ãºltimos 5 segundos (extensiÃ³n automÃ¡tica)
4. El precio de reserva es opcional pero debe ser â‰¥ precio inicial
5. Las pujas automÃ¡ticas deben respetar los mismos lÃ­mites

### ParticipaciÃ³n
1. Un vendedor no puede pujar en su propia subasta
2. Un usuario bloqueado no puede participar
3. Se requiere balance suficiente para el depÃ³sito
4. MÃ¡ximo 50 pujas por usuario en una subasta
5. No se permiten pujas despuÃ©s de la fecha de fin

## ğŸ›¡ï¸ Validaciones

### Pre-Subasta
```typescript
interface PreAuctionChecks {
    validateProduct(): boolean;
    validatePricing(): boolean;
    validateDates(): boolean;
    validateSeller(): boolean;
}
```

### Durante Subasta
```typescript
interface ActiveAuctionChecks {
    validateBid(bid: Bid): boolean;
    checkTimeExtension(): boolean;
    verifyUserEligibility(user: User): boolean;
    validatePauseRequest(): boolean;
}
```

### Post-Subasta
```typescript
interface PostAuctionChecks {
    validateWinner(): boolean;
    verifyPaymentEligibility(): boolean;
    checkShippingRequirements(): boolean;
}
```

## ğŸ¯ Comportamiento EspecÃ­fico

### ExtensiÃ³n AutomÃ¡tica
1. Si hay una puja en los Ãºltimos 2 minutos
2. Se extiende 2 minutos desde la Ãºltima puja
3. MÃ¡ximo 5 extensiones por subasta
4. Se notifica a todos los participantes

### Pausas
1. MÃ¡ximo 2 pausas por subasta
2. DuraciÃ³n mÃ¡xima de 24 horas
3. No permitido en las Ãºltimas 12 horas
4. Requiere justificaciÃ³n vÃ¡lida

### CancelaciÃ³n
1. Solo permitida si no hay pujas
2. PenalizaciÃ³n si es injustificada
3. Requiere aprobaciÃ³n si hay pujas
4. Se devuelven todos los depÃ³sitos

## ğŸ“Š CÃ¡lculos y FÃ³rmulas

### DepÃ³sitos
```typescript
interface DepositCalculations {
    minDeposit = currentPrice * 0.05;
    maxDeposit = currentPrice * 0.20;
    requiredDeposit = max(minDeposit, customDeposit);
}
```

### Incrementos
```typescript
interface BidIncrements {
    minIncrement = currentPrice * 0.01;
    suggestedIncrements = [
        currentPrice * 0.01,
        currentPrice * 0.02,
        currentPrice * 0.05
    ];
}
```

## ğŸ”„ Estados y Transiciones

### Estados Permitidos
```
[Borrador] â†’ [Programada] â†’ [Activa] â†’ [Finalizada]
                â†‘   â†“           â†“
              [Pausada]     [Cancelada]
```

### Condiciones de TransiciÃ³n
1. Borrador â†’ Programada: ValidaciÃ³n completa
2. Programada â†’ Activa: Fecha inicio alcanzada
3. Activa â†’ Pausada: Solicitud aprobada
4. Activa â†’ Finalizada: Tiempo completado
5. Activa â†’ Cancelada: Condiciones cumplidas

## ğŸš« Restricciones

### Productos
1. No permitidos:
   - ArtÃ­culos ilegales
   - Productos falsificados
   - ArtÃ­culos peligrosos
   - Servicios personales

### Usuarios
1. LÃ­mites por usuario:
   - MÃ¡ximo 10 subastas activas
   - MÃ¡ximo 50 pujas activas
   - MÃ­nimo 90% rating positivo

### Tiempos
1. Restricciones temporales:
   - MÃ­nimo 1 dÃ­a
   - MÃ¡ximo 30 dÃ­as
   - No iniciar en festivos
   - Horario comercial preferido

## ğŸ“± Notificaciones

### Eventos Notificables
1. Inicio de subasta
2. Nueva puja
3. Superado en puja
4. PrÃ³ximo a finalizar
5. FinalizaciÃ³n
6. Ganador
7. Cambios de estado

### Canales
1. Prioridad por tipo:
   - CrÃ­ticas: Email + Push + SMS
   - Importantes: Email + Push
   - Informativas: Push
