# üì¶ Reglas de Productos

## üìù Reglas Generales

### Creaci√≥n
1. Informaci√≥n B√°sica
   - T√≠tulo descriptivo (10-100 chars)
   - Descripci√≥n detallada
   - Categor√≠a principal
   - Subcategor√≠as (max 3)

2. Media
   - M√≠nimo 3 im√°genes
   - Primera imagen destacada
   - Calidad m√≠nima 800x800
   - M√°ximo 10 im√°genes

3. Precios
   - Valor de mercado
   - Precio m√≠nimo
   - Precio sugerido
   - Incrementos recomendados

### Validaci√≥n
1. Contenido
   - No contenido ofensivo
   - Informaci√≥n precisa
   - Sin datos personales
   - Links permitidos limitados

2. Calidad
   - Fotos claras
   - Descripci√≥n completa
   - Especificaciones t√©cnicas
   - Estado del producto

## üõ°Ô∏è Validaciones T√©cnicas

### Producto
```typescript
interface ProductValidation {
    validateBasicInfo(): boolean;
    checkMedia(): boolean;
    validatePricing(): boolean;
    verifyCategory(): boolean;
    checkRestrictions(): boolean;
}
```

### Media
```typescript
interface MediaValidation {
    checkImageQuality(): boolean;
    validateFileTypes(): boolean;
    scanForInappropriate(): boolean;
    verifyDimensions(): boolean;
}
```

## üéØ Categorizaci√≥n

### Jerarqu√≠a
1. Categor√≠a Principal
   - √önica
   - Obligatoria
   - De lista predefinida
   - Con metadata espec√≠fica

2. Subcategor√≠as
   - M√°ximo 3
   - Relacionadas
   - Coherentes
   - Indexables

### Atributos
```typescript
interface CategoryAttributes {
    required: string[];
    optional: string[];
    specific: Record<string, any>;
    validation: Record<string, any>;
}
```

## üìä Precios y Valores

### C√°lculos
```typescript
interface PriceCalculations {
    marketValue: number;
    minPrice = marketValue * 0.5;
    suggestedPrice = marketValue * 0.8;
    maxPrice = marketValue * 1.5;
    
    calculateIncrement(): number[] {
        return [
            marketValue * 0.01,
            marketValue * 0.02,
            marketValue * 0.05
        ];
    }
}
```

### Restricciones
1. Por Categor√≠a
   - Rango permitido
   - Incrementos m√≠nimos
   - Valores m√°ximos
   - Dep√≥sitos requeridos

2. Por Estado
   - Nuevo vs Usado
   - Coleccionable
   - Edici√≥n limitada
   - Vintage

## üîÑ Estados de Producto

### Ciclo de Vida
```
[Borrador] ‚Üí [En Revisi√≥n] ‚Üí [Activo]
     ‚Üì             ‚Üì            ‚Üì
[Descartado]  [Rechazado]   [Pausado] ‚Üí [Vendido]
```

### Transiciones
1. Borrador ‚Üí Revisi√≥n
   - Info completa
   - Media v√°lida
   - Categor√≠a correcta
   - Precio v√°lido

2. Revisi√≥n ‚Üí Activo
   - Moderaci√≥n aprobada
   - Vendedor verificado
   - Sin restricciones
   - Documentaci√≥n OK

## üö´ Restricciones

### Productos Prohibidos
1. Categor√≠as
   - Ilegales
   - Peligrosos
   - Falsificados
   - Restringidos

2. Condiciones
   - Da√±ados
   - Incompletos
   - Sin garant√≠a
   - Sin procedencia

### L√≠mites
1. Por Usuario
   - M√°ximo 50 activos
   - 10 por categor√≠a
   - 5 destacados
   - 3 premium

2. Por Producto
   - 10 im√°genes
   - 5000 chars descripci√≥n
   - 20 tags
   - 3 subcategor√≠as

## üì± Notificaciones

### Al Vendedor
1. Creaci√≥n
   - Confirmaci√≥n
   - Sugerencias
   - Mejoras
   - Estado

2. Moderaci√≥n
   - Aprobaci√≥n
   - Rechazo
   - Cambios requeridos
   - Avisos

### Al Sistema
1. Revisi√≥n
   - Nuevos productos
   - Modificaciones
   - Reportes
   - Duplicados

2. Alertas
   - Contenido inadecuado
   - Precios an√≥malos
   - Patrones sospechosos
   - Infracciones

## üîç SEO y Visibilidad

### Optimizaci√≥n
1. T√≠tulos
   - Keywords relevantes
   - Estructura clara
   - Sin repetici√≥n
   - Longitud √≥ptima

2. Descripci√≥n
   - Formato rico
   - Puntos clave
   - Especificaciones
   - FAQs incluidas

### Metadata
```typescript
interface ProductMetadata {
    seoTitle: string;
    description: string;
    keywords: string[];
    schema: Record<string, any>;
    
    generateRichSnippets(): string {
        // Generaci√≥n de rich snippets
    }
}
```

## üìä Analytics

### M√©tricas
1. Rendimiento
   - Vistas
   - Inter√©s
   - Conversi√≥n
   - Tiempo activo

2. Calidad
   - Score de completitud
   - Rating de fotos
   - Precisi√≥n de datos
   - Relevancia

### Reportes
1. Por Producto
   - Performance
   - Interacciones
   - Comparativas
   - Tendencias

2. Agregados
   - Categor√≠as top
   - Precios promedio
   - Tasas de √©xito
   - Patrones
