# â­ Reglas de Valoraciones y Comentarios

## ğŸ“ Reglas Generales

### Valoraciones
1. Solo despuÃ©s de transacciÃ³n completada
2. PerÃ­odo lÃ­mite: 30 dÃ­as post-entrega
3. Una valoraciÃ³n por transacciÃ³n
4. Modificable hasta 7 dÃ­as despuÃ©s
5. Incluye rating (1-5) y comentario

### Comentarios
1. Longitud: 10-1000 caracteres
2. No contenido ofensivo/spam
3. Relevante a la transacciÃ³n
4. ModeraciÃ³n antes de publicaciÃ³n
5. Respuesta Ãºnica del vendedor

## ğŸ›¡ï¸ Validaciones

### Valoraciones
```typescript
interface RatingValidation {
    validateTransaction(): boolean;
    checkTimeWindow(): boolean;
    validateScore(): boolean;
    checkDuplication(): boolean;
    validateContent(): boolean;
}
```

### Comentarios
```typescript
interface CommentValidation {
    checkLength(): boolean;
    validateContent(): boolean;
    checkSpam(): boolean;
    validateUser(): boolean;
    checkModeration(): boolean;
}
```

## ğŸ¯ Sistema de PuntuaciÃ³n

### CÃ¡lculo de Rating
```typescript
interface RatingCalculation {
    userScore = weightedAverage([
        lastMonth: 0.5,
        last3Months: 0.3,
        allTime: 0.2
    ]);
    
    productScore = weightedAverage([
        quality: 0.4,
        accuracy: 0.3,
        communication: 0.3
    ]);
}
```

### Impacto
1. Visibilidad en bÃºsquedas
2. LÃ­mites de operaciÃ³n
3. Badges y reconocimientos
4. Acceso a caracterÃ­sticas premium

## ğŸ“Š ModeraciÃ³n

### Filtros AutomÃ¡ticos
1. Palabras prohibidas
2. Patrones de spam
3. Contenido duplicado
4. Enlaces externos
5. InformaciÃ³n personal

### RevisiÃ³n Manual
1. Contenido flaggeado
2. Valoraciones extremas
3. Reportes de usuarios
4. Patrones sospechosos
5. Apelaciones

## ğŸ”„ Estados

### Valoraciones
```
[Pendiente] â†’ [En RevisiÃ³n] â†’ [Publicada]
     â†“             â†“             â†“
[Expirada]    [Rechazada]    [Eliminada]
```

### Comentarios
```
[Borrador] â†’ [En ModeraciÃ³n] â†’ [Publicado]
     â†“              â†“             â†“
[Cancelado]     [Rechazado]   [Archivado]
```

## ğŸš« Restricciones

### Contenido Prohibido
1. Lenguaje ofensivo
2. InformaciÃ³n personal
3. Spam/Publicidad
4. Enlaces externos
5. Amenazas/Acoso

### LÃ­mites
1. Por Usuario
   - Max 3 valoraciones/dÃ­a
   - Max 10 comentarios/dÃ­a
   - Min 24h entre valoraciones
   - Max 5 ediciones totales

2. Por Item
   - Max 1 valoraciÃ³n/compra
   - Max 1 respuesta/valoraciÃ³n
   - Max 100 comentarios/item
   - Max 3 reportes antes de revisiÃ³n

## ğŸ“± Notificaciones

### Al Autor
1. ValoraciÃ³n publicada
2. Comentario aprobado
3. Contenido rechazado
4. Respuesta recibida
5. Reporte procesado

### Al Receptor
1. Nueva valoraciÃ³n
2. Nuevo comentario
3. Respuesta requerida
4. Alerta de reporte
5. Recordatorio pendiente

## ğŸŒŸ GamificaciÃ³n

### Badges
1. Top Reviewer
   - 50+ valoraciones
   - 90%+ Ãºtiles
   - Contenido detallado
   - Sin rechazos

2. Helpful Commenter
   - 100+ comentarios
   - Alta interacciÃ³n
   - Respuestas Ãºtiles
   - Contribuciones valoradas

### Incentivos
1. Puntos por participaciÃ³n
2. Descuentos especiales
3. Acceso anticipado
4. Badges exclusivos
5. Privilegios especiales

## ğŸ“Š AnÃ¡lisis

### MÃ©tricas
1. SatisfacciÃ³n
   - Rating promedio
   - Tendencias
   - DistribuciÃ³n
   - Comparativas

2. Engagement
   - Tasa de respuesta
   - Tiempo promedio
   - Interacciones
   - Utilidad percibida

### Reportes
1. Diarios
   - Nuevas valoraciones
   - ModeraciÃ³n pendiente
   - Reportes activos
   - Tendencias

2. Mensuales
   - AnÃ¡lisis de sentimiento
   - Patrones de uso
   - Ãreas de mejora
   - Recomendaciones

## ğŸ” SEO y Visibilidad

### OptimizaciÃ³n
1. Estructura rica
   - Microdata
   - Schema.org
   - Rich snippets
   - Agregaciones

2. IndexaciÃ³n
   - Prioridad por relevancia
   - ActualizaciÃ³n frecuente
   - Filtrado de spam
   - Contenido Ãºnico
